packages/rest-client/src/main/types/Errors.ts

```diff
@@ -0,0 +1,12 @@
+import type { ResponseError } from 'superagent'
+
+export interface SanitisedError extends Error {
+  text?: string
+  status?: number
+  headers?: unknown
+  data?: unknown
```

Comment:
in our projects, we make this type generic so that the `data` property can be specified. would you be interested in seeing that version? or me adding it here?

Reply:
i think similarly it'd be nice to parameterise the _request_ body type

Reply:
rather than have an `unknown` type, one can do something like
https://github.com/ministryofjustice/hmpps-incident-reporting/blob/cb9c8e08ec81084d6ebc17d29b6e97f46f56986f/server/sanitisedError.ts#L6-L14

…because our apis often have a well-defined error response structure and you might want to pull out, i dunno, some special code

Reply:
Hmm, tempting - though I think it's best to keep this generic for now as I imagine people might use the RestClient to call stuff outside of HMPPS occasionally too

Reply:
my suggestion would in fact make it generic! right now, we all get `unknown` but the *optional* generic type in my link above allows consumers of this lib to override the type

Reply:
Oooh I follow! Let me get that added, I like that :)

Reply:
Added in https://github.com/ministryofjustice/hmpps-typescript-lib/pull/54/commits/da1fb113cc162939fdfca1090e3e403fb9e09b8c